modules/targets/darwin/options.nix

```diff
@@ -63,6 +73,280 @@ in {
+        description = ''
+          Whether to use expanded save panel by default.  The default is false.
+        '';
+      };
+
+      NSTableViewDefaultSizeMode = mkOption {
+        type = types.nullOr (types.enum [ 1 2 3 ]);
+        default = null;
+        description = ''
+          Sets the size of the finder sidebar icons: 1 (small), 2 (medium) or 3 (large). The default is 3.
+        '';
+      };
+
+      NSTextShowsControlCharacters = mkOption {
+        type = types.nullOr types.bool;
+        default = null;
+        description = ''
+          Whether to display ASCII control characters using caret notation in standard text views. The default is false.
+        '';
+      };
+
+      NSUseAnimatedFocusRing = mkOption {
+        type = types.nullOr types.bool;
+        default = null;
+        description = ''
+          Whether to enable the focus ring animation. The default is true.
+        '';
+      };
+
+      NSScrollAnimationEnabled = mkOption {
+        type = types.nullOr types.bool;
+        default = null;
+        description = ''
+          Whether to enable smooth scrolling. The default is true.
+        '';
+      };
+
+      NSWindowResizeTime = mkOption {
+        type = types.nullOr floatStr;
+        default = null;
+        example = "0.20";
+        description = ''
+          Sets the speed speed of window resizing. The default is given in the example.
+        '';
+      };
+
+      InitialKeyRepeat = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = ''
+          # Apple menu > System Preferences > Keyboard
+          If you press and hold certain keyboard keys when in a text area, the key’s character begins to repeat.
+          For example, the Delete key continues to remove text for as long as you hold it down.
+
+          This sets how long you must hold down the key before it starts repeating.
+        '';
+      };
+
+      KeyRepeat = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = ''
+          # Apple menu > System Preferences > Keyboard
+          If you press and hold certain keyboard keys when in a text area, the key’s character begins to repeat.
+          For example, the Delete key continues to remove text for as long as you hold it down.
+
+          This sets how fast it repeats once it starts.
+        '';
+      };
+
+      PMPrintingExpandedStateForPrint = mkOption {
+        type = types.nullOr types.bool;
+        default = null;
+        description = ''
+          Whether to use the expanded print panel by default. The default is false.
+        '';
+      };
+
+      PMPrintingExpandedStateForPrint2 = mkOption {
+        type = types.nullOr types.bool;
+        default = null;
+        description = ''
+          Whether to use the expanded print panel by default. The default is false.
+        '';
+      };
+
+      "com.apple.keyboard.fnState" = mkOption {
```

Comment:
It kind of surprised me that this option belongs to `NSGlobalDomain` and not in its own domain, `com.apple.keyboard`. It may be worth adding a code comment explaining that this option is placed where it's supposed to be.